opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java

```diff
@@ -38,16 +42,26 @@
 public class MockTracer implements Tracer {
     private List<MockSpan> finishedSpans = new ArrayList<>();
     private final Propagator propagator;
+    private ActiveSpanSource spanSource;
 
     public MockTracer() {
         this(Propagator.PRINTER);
     }
 
+    public MockTracer(ActiveSpanSource spanSource) {
+        this(spanSource, Propagator.PRINTER);
+    }
+
+    public MockTracer(ActiveSpanSource spanSource, Propagator propagator) {
+        this.propagator = propagator;
+        this.spanSource = spanSource;
+    }
+
     /**
      * Create a new MockTracer that passes through any calls to inject() and/or extract().
      */
     public MockTracer(Propagator propagator) {
-        this.propagator = propagator;
+        this(NoopActiveSpanSource.INSTANCE, propagator);
```

Comment:
@pavolloffay you mean instead of a noop? This runs into problems with circular dependencies since the ThreadLocalActiveSpanSourceTest depends on the MockTracer. We allow the programmer to manually specify an ActiveSpanSource via an overloaded constructor; I thought that was sufficient.